### ISSUE TITLE
Submit to AUR 

### ISSUE THREAD

**Issue by curusarn** [curusarn/resh#73]
https://wiki.archlinux.org/index.php/AUR_submission_guidelines

**Comment by curusarn** [comment]
Way too many people asked me to put this on AUR so I will look into it soon.

**Comment by deathtrip** [comment]
any progress on this?

**Comment by Roboron3042** [comment]
It seems that the current installation consists on installing resh and its configuration files to the home directory of the current user, and it also needs to modify zshrc. This is not what a package should do, so it's not an easy task.

I would say that it is not worth the hassle since you can easily install resh with just a command in any distro, anyway.

**Comment by deathtrip** [comment]
You can give the user a notice to run a script, that will modify zshrc, after installation.
Having a package in the AUR, makes it much easier to update software.

**Comment by Roboron3042** [comment]
Or you can just add `reshctl update` to your zshrc file and it will look for updates when you open your shell. Easier than that...

**Comment by deathtrip** [comment]
No, it's not easier. Updating packages by adding commands to zshrc is an ugly hack. It slows down shell startup and needlessly hammers servers. There's a reason people developed package managers.

**Comment by Roboron3042** [comment]
> No, it's not easier. 

Yes it is. It's not efficient, but easy =/= efficient.

> It slows down shell startup and needlessly hammers servers. 

You just discovered how self-updated applications and update software centres work.

> There's a reason people developed package managers.

And that reason is not to ask the user to run a install script. If you only concern is to update along the rest of the system, just run `reshctl update` after a `pacman -Syu`. You could even create an alias `update` which does both, or an script, etc...

**Comment by deathtrip** [comment]
You need to modify zshrc to use any zsh plugin, even from the repos or AUR. So it's not an "install script".
If i had to create aliases, run update scripts etc. for every package not in the official repos, it would just be a one big mess.

**Comment by curusarn** [comment]
Sorry about the long silence, I was busy graduating during the past few months.

I do think that the current way to update is fairly convenient but it is definitely not ideal.

Installation using a package manager is superior (mainly because of the ability to conveniently update all software at the same time). Because of that, I want to support it eventually.

That said it's not very easy to package RESH because of the necessary integration with user's shell. This is why I haven't gotten around to publish RESH to AUR yet.

There are multiple connected issues that make a lot of sense to solve before (or at the same time as) packaging. 

Specifically, I would like to move the RESH installation outside of the user's HOME. (This is important because system package manager shouldn't install things into user's HOME)

And before doing that, I need to create some automatised tests to make sure that I'm not breaking things as I make changes to the RESH install process.

I know this is probably not what you wanted to hear. I hope you understand that I don't want to hack together a submission to AUR that will randomly fail in the future.

I wish I could give you a more specific timeframe. I haven't forgotten about AUR and I haven't changed my mind.

For the past few months I have been working on the core functionality of RESH. Now I want to make it more stable and publishing RESH to AUR is a part of that. 

Thank you for your patience.

TL;DR Publishing to AUR is still on my radar but it makes more sense to do it a bit later as a part of other changes

**Comment by curusarn** [comment]
Small update to this issue.

No promises but in upcoming release (#184) Resh files got more organised so it should be easier to release it to AUR, Homebrew, etc.

I just want things to be well thought out so that supporting different installation ways doesn't come with a lot of bugs and too much maintainance effort.
